FirebaseAppDistribution/Sources/FIRAppDistribution.m

```diff
@@ -0,0 +1,270 @@
+                                     presentingViewController:self.safariHostingViewController
+                                                     callback:^(OIDAuthState *_Nullable authState,
+                                                                NSError *_Nullable error) {
+                                                       self.authState = authState;
+                                                       self.isTesterSignedIn =
+                                                           self.authState ? YES : NO;
+                                                       completion(error);
+                                                     }];
+}
+
+- (UIWindow *)createUIWindowForLogin {
+  // Create an empty window + viewController to host the Safari UI.
+  UIWindow *window = [[UIWindow alloc] initWithFrame:[UIScreen mainScreen].bounds];
+  window.rootViewController = self.safariHostingViewController;
+
+  // Place it at the highest level within the stack.
+  window.windowLevel = +CGFLOAT_MAX;
+
+  // Run it.
+  [window makeKeyAndVisible];
+
+  return window;
+}
+
+- (void)handleReleasesAPIResponseWithData:data
+                               completion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  // TODO Implement parsing of releases API response
+  completion(nil, nil);
+}
+
+- (void)checkForUpdateWithCompletion:(FIRAppDistributionUpdateCheckCompletion)completion {
+  if (self.isTesterSignedIn) {
+    [self fetchReleases:completion];
+  } else {
+    UIAlertController *alert = [UIAlertController
+        alertControllerWithTitle:@"Enable in-app alerts"
+                         message:@"Sign in with your Firebase App Distribution Google account to "
+                                 @"turn on in-app alerts for new test releases."
+                  preferredStyle:UIAlertControllerStyleAlert];
+
+    UIAlertAction *yesButton =
+        [UIAlertAction actionWithTitle:@"Turn on"
+                                 style:UIAlertActionStyleDefault
+                               handler:^(UIAlertAction *action) {
+                                 [self signInTesterWithCompletion:^(NSError *_Nullable error) {
+                                   self.window.hidden = YES;
+                                   self.window = nil;
+
+                                   if (error) {
+                                     completion(nil, error);
+                                     return;
+                                   }
+
+                                   [self fetchReleases:completion];
+                                 }];
+                               }];
+
+    UIAlertAction *noButton = [UIAlertAction actionWithTitle:@"Not now"
+                                                       style:UIAlertActionStyleDefault
+                                                     handler:^(UIAlertAction *action) {
+                                                       // precaution to ensure window gets destroyed
```

Comment:
"Precaution that ensures window gets destroyed." To be consistent with other comments.